### PR TITLE
Enable Docker for Llama.cpp Benchmarking

### PR DESCRIPTION
**Pull request by zxybazh**
Add docker scripts to benchmark llama.cpp performance for llm models on cuda.

### CODE REVIEW

**Comment by zxybazh** [comment]
Ready for review! CC: @junrushao

**Comment by junrushao** [comment]
Thanks @zxybazh! Will review tomorrow by trying to actually run it on my 3090ti :)

File: Dockerfile.cu121.llama.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Step 3. Git clone and compile llama.cpp with cuBLAS
+RUN git clone https://github.com/ggerganov/llama.cpp.git $LLAMA_CPP_HOME && \
+    cd llama.cpp                                                         && \
+    git checkout f3c3b4b1672d860800639c87d3b5d17564692469                && \
+    make LLAMA_CUBLAS=1                                                  && \
+    mkdir build                                                          && \
+    cd build                                                             && \
+    cmake .. -DLLAMA_CUBLAS=1                                            && \
+    cmake --build . --config Release
```
Comment (junrushao) [inline Dockerfile.cu121.llama.cpp]: not super familiar, just wanted to double check - are we actually building llama.cpp twice? cause i saw the line `make LLMA_CUBLAS=1` as well as the `cmake --build`

File: README.md [hunk 2]
```diff
@@ -73,7 +74,43 @@ TBD
...
+```
+
+**Step 3.** Run the CLI tool to see the performance numbers:
+
+GGML binary files are like `llama-2-7b.ggmlv3.q4_K_M.bin` or `llama-2-7b.ggmlv3.q4_0.bin`.
+You can download llama-2 7B GGML binary files from [here](https://huggingface.co/TheBloke/Llama-2-7B-GGML/tree/main).
+
+```bash
+cd $LLAMA_CPP_HOME
+./build/bin/main -m /ggml_bins/GGML_BINARY_FILE -p "Please generate a very long story about wizard and technology, at least two thousand words" -n 1920 -ngl 999 --ignore-eos |& tee llama_cpp_result.log
```
Comment (junrushao) [inline README.md]: I followed this instruction but it generates gibberish. Removing `-ngl 999` does make it work and generate sane stuff, but it's incredibly slow

Comment (junrushao) [reply README.md]: well interesting...the `main` from Makefile generates gibberish, but the one from cmake doesn't...

File: Dockerfile.cu121.llama.cpp (hunk 1 above)
Comment (zxybazh) [reply Dockerfile.cu121.llama.cpp]: Thanks for pointing this out, yes I accidentally copied this line, I intend to only use `cmake` here.
Reference: https://github.com/ggerganov/llama.cpp/tree/master#cublas

File: README.md (hunk 2 above)
Comment (zxybazh) [reply README.md]: Yes, I was referring to the cmake results, somehow the make result doesn't generate anything sane.

File: README.md [hunk 3]
```diff
@@ -73,7 +74,47 @@ TBD
...
+ssh root@0.0.0.0 -p $PORT
+```
+
+**Step 3.** Run the CLI tool to see the performance numbers:
+
+Log in to the docker container we created using the comamnd below:
+
+```bash
+cd $LLAMA_CPP_HOME
+./build/bin/main -m /ggml_bins/llama-2-7b.ggmlv3.q4_K_M.bin -p "Please generate a very long story about wizard and technology, at least two thousand words" -n 1920 -ngl 999 --ignore-eos |& tee llama_cpp_result.log
```
Comment (zxybazh) [inline README.md]: Note that we are using a pretty short prompt here, may be replaced with a standard length promot if necessary. In MLC-LLM we are using randomly generated input with given input length.

**Review by junrushao — APPROVED** [review]

**Pull request merged**